### ISSUE TITLE
[Nested Tensor] Dropout(0.0) increments cuda rng_state (only for no-compile)

### ISSUE THREAD

**Issue by mikaylagawarecki** [pytorch/pytorch#136656]
### 🐛 Describe the bug

This only happens without compile, so might be more of an edge case.

I ran into this while trying to implement a causal self attn + ffn with NJT and validating the results (the rng_state increment here somehow seemed to affect the output, can share a larger repro if interesting).

Conventionally running `Dropout(0.0)` should not bump cuda rng state

```python
import torch
import torch.nn as nn

m = nn.Dropout(0.0)
inp = torch.randn(2, 3, device='cuda')
gpu_rng_state = torch.cuda.get_rng_state()
m(inp)
assert torch.equal(torch.cuda.get_rng_state(), gpu_rng_state)  # passes



m = nn.Dropout(0.0)
sentence_lengths = torch.tensor([2, 3], device='cuda')
nested_inp = torch.nested.nested_tensor([
                torch.randn(l.item(), 3, device='cuda')
                for l in sentence_lengths
            ], layout=torch.jagged)
compiled_m = torch.compile(m)
gpu_rng_state = torch.cuda.get_rng_state()
compiled_m(nested_inp)
assert torch.equal(torch.cuda.get_rng_state(), gpu_rng_state) # passes



m = nn.Dropout(0.0)
sentence_lengths = torch.tensor([2, 3], device='cuda')
nested_inp = torch.nested.nested_tensor([
                torch.randn(l.item(), 3, device='cuda')
                for l in sentence_lengths
            ], layout=torch.jagged)
gpu_rng_state = torch.cuda.get_rng_state()
m(nested_inp)
assert torch.equal(torch.cuda.get_rng_state(), gpu_rng_state)  # fails
```

### Versions

2.6.0.dev20240925+cu124
